keras_cv/src/models/object_detection/faster_rcnn/faster_rcnn.py

```diff
@@ -0,0 +1,477 @@
+                    "`rpn_classification_loss.from_logits="
+                    f"{classification_loss.from_logits}`"
+                )
+        if hasattr(classification_loss, "from_logits"):
+            if not classification_loss.from_logits:
+                raise ValueError(
+                    "FasterRCNN.compile() expects `from_logits` to be True for "
+                    "`classification_loss`. Got "
+                    "`classification_loss.from_logits="
+                    f"{classification_loss.from_logits}`"
+                )
+        if hasattr(box_loss, "bounding_box_format"):
+            if box_loss.bounding_box_format != self.bounding_box_format:
+                raise ValueError(
+                    "Wrong `bounding_box_format` passed to `box_loss` in "
+                    "`FasterRCNN.compile()`. Got "
+                    "`box_loss.bounding_box_format="
+                    f"{box_loss.bounding_box_format}`, want "
+                    "`box_loss.bounding_box_format="
+                    f"{self.bounding_box_format}`"
+                )
+        self.rpn_box_loss = rpn_box_loss
+        self.rpn_cls_loss = rpn_classification_loss
+        self.box_loss = box_loss
+        self.cls_loss = classification_loss
+        self.weight_decay = weight_decay
+        losses = {
+            "box": self.box_loss,
+            "classification": self.cls_loss,
+            "rpn_box": self.rpn_box_loss,
+            "rpn_classification": self.rpn_cls_loss,
+        }
+        self._has_user_metrics = metrics is not None and len(metrics) != 0
+        self._user_metrics = metrics
+        super().compile(loss=losses, **kwargs)
+        
+    def compute_loss(self, x, y, y_pred, sample_weight, **kwargs):
+        # 1. Unpack the inputs
+        images = x
+        gt_boxes = y["boxes"]
+        if keras.ops.ndim(y["classes"]) != 2:
+            raise ValueError(
+                "Expected 'classes' to be a Tensor of rank 2. "
+                f"Got y['classes'].shape={keras.ops.shape(y['classes'])}."
+            )
+            
+        gt_classes = y["classes"]
+        gt_classes = keras.ops.expand_dims(y["classes"], axis=-1)
+        
+        # Generate anchors
+        # image shape must not contain the batch size
+        local_batch = keras.ops.shape(images)[0]
+        image_shape = keras.ops.shape(images)[1:]
+        anchors = self.anchor_generator(image_shape=image_shape)
+        
+        # 2. Label with the anchors -- exclusive to compute_loss
+        (
+            rpn_box_targets,
+            rpn_box_weights,
+            rpn_cls_targets,
+            rpn_cls_weights,
+        ) = self.rpn_labeler(
+            anchors_dict=keras.ops.concatenate(
+                tree.flatten(anchors),
+                axis=0,
+            ),
+            gt_boxes=gt_boxes,
+            gt_classes=gt_classes,
+        )
+        # 3. Computing the weights
+        rpn_box_weights /= (
```

Comment:
Here for cls/box weights we need batch size and while testing I observed that it failes when given odd batch size as batch size becomes `None`. Not sure what is the causing the probelm.

The dataset constructed should be from tensor slices striagt forward `model.fit(xs, ys, epochs=1)` seems to fail as well.

![image](https://github.com/keras-team/keras-cv/assets/113718461/3987e068-be43-4739-a23e-890ded79f74d)

Reply:
I would agree that we probably do need batch_size. I've never really seen it removed in operations like this before.